### PR TITLE
[Webtoons] - Initial Implementation

### PR DESCRIPTION
**Pull request by ConradWeiser**
Manga Updates is not available

### CODE REVIEW

**Comment by ConradWeiser** [comment]
Searches do not work if there are spaces.
`" "` --> `"%20"` needs slapped in there for search titles.

File: src/Webtoons/Webtoons.ts [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+      const $ = this.cheerio.load(data.data)
+      return parseChapterDetailsOrig($, mangaId, chapterId)
+    }
+
+  }
+
+  async searchRequest(query: SearchRequest, metadata: any): Promise<PagedResults> {
+    const request = createRequestObject({
+      url: `${WEBTOON_DOMAIN}/en/search?keyword=${query.title}`,
```
Comment (nar1n) [inline src/Webtoons/Webtoons.ts:218]:
```suggestion
      url: `${WEBTOON_DOMAIN}/en/search?keyword=${query.title?.replace(/ /g, '%20')}`,
```

**Review by nar1n — CHANGES_REQUESTED** [review]

**Review by nar1n — APPROVED** [review]

**Comment by ConradWeiser** [comment]
It's been reported that all Orig. series show as 'Completed' even if that's not the case. Something to look into.
Uses: `    const completed = $("span.txt_ico_completed2") === undefined ? MangaStatus.ONGOING : MangaStatus.COMPLETED;`

Targets: 
image

**Review by nar1n — APPROVED** [review]

File: src/Webtoons/WebtoonsParser.ts [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+    // The image is awkwardly embedded into a style, parse that out
+    const imageContext = $("div.detail_body").attr("style");
+    const image = imageContext?.match(/url\((.+)\)/);
+
+    if (!imageContext || !image || !image[1]) {
+      throw new Error(`Failed to parse MangaDetails for ${mangaId}`);
+    }
+
+    let rating = $("em#_starScoreAverage").text();
+    const completed = $("span.txt_ico_completed2") === undefined ? MangaStatus.ONGOING : MangaStatus.COMPLETED;
```
Comment (nar1n) [inline src/Webtoons/WebtoonsParser.ts:46]:
```suggestion
    const completed = $("p.day_info").text().includes('COMPLETED') ? MangaStatus.COMPLETED : MangaStatus.ONGOING;
```
Tested and this works

File: src/Webtoons/WebtoonsParser.ts [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+const title = $('h3._challengeTitle').text().replace('DASHBOARD', '').trim()
+
+const image = $('img', $('span.thmb')).attr('src')
+
+if(!image) {
+    throw new Error(`Failed to parse MangaDetails challenge image for ${mangaId}`)
+}
+
+let rating = $('em#_starScoreAverage').text()
+const completed = $('span.txt_ico_completed2') === undefined ? MangaStatus.ONGOING : MangaStatus.COMPLETED
```
Comment (nar1n) [inline src/Webtoons/WebtoonsParser.ts:15]:
```suggestion
const completed = title.toLowerCase().includes('completed') ? MangaStatus.COMPLETED : MangaStatus.ONGOING
```
Completed manhwa are marked in title

**Review by nar1n — CHANGES_REQUESTED** [review]

**Review by nar1n — APPROVED** [review]

**Review by TheNetsky — APPROVED** [review]

**Pull request merged**